### PR TITLE
Setup release management

### PR DESCRIPTION
**Pull request by vzhukovs**
This changes proposal adds release management to the repository.

It generates images structure according to https://github.com/eclipse/che/issues/19698

fixes https://github.com/eclipse/che/issues/19698

- **che-idea** (_vzhukovs_ repository is for test):
Снимок экрана 2021-06-24 в 11 29 57

- **che-pycharm** (_vzhukovs_ repository is for test):
Снимок экрана 2021-06-24 в 11 31 47

For each commit in master branch, there is a particular image with:

- `che-idea:version-next`
- `che-pycharm:version-next`

---

There are two new scripts:

- `projector.sh`
  - Performs all routine work with building and running docker images
- `make-release.sh`
  - Performs the release of particular version provided by `--tag` parameter. If it is omitted, then default tag name selected in format `YYYYMMDD.hashId`
  - Publishes the released artifacts to quay.io
  - Creates the automatic draft release on GitHub and publishes Docker images to GitHub release artifacts. See screenshot below. These docker images can be loaded by `$ docker load` command.

Снимок экрана 2021-06-24 в 11 41 14

---

**Action required.**
Before merging, repository should contains secret `QUAY_REPOSITORY` with value `quay.io/che-incubator`. This is done to allow forks to have the ability to perform GitHub actions on their repository.

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### CODE REVIEW

**Comment by vzhukovs** [comment]
PR check might fail due to missed secret `QUAY_REPOSITORY`

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    steps:
+      - name: Clone github repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - name: Tag name
+        id: tag_name
+        run: echo ::set-output name=sourceTag::${GITHUB_REF#refs/tags/}
+      - name: Docker login
+        uses: azure/docker-login@v1
```
Comment (benoitf) [inline .github/workflows/release.yml:63]: I would use the docker action
https://github.com/docker/login-action

File: compatible-ide.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+[
+  {
+    "displayName": "IntelliJ IDEA Community",
+    "dockerImage": "che-idea",
+    "productCode": "IC",
+    "productVersion": [
+      {
+        "version": "2020.3.3",
+        "downloadUrl": "https://download.jetbrains.com/idea/ideaIC-2020.3.3.tar.gz",
```
Comment (benoitf) [inline compatible-ide.json:9]: Not related to this PR, side question
Do we plan to use 2021 binaries soin ?

**Review by benoitf — APPROVED** [review]

File: compatible-ide.json (hunk 2 above)
Comment (vzhukovs) [reply compatible-ide.json:9]: 2021 is not supported yet by Projector Server, there are some incompatibility with Markdown plugin AFAIK.
There is an upstream issue for that https://youtrack.jetbrains.com/issue/PRJ-547#focus=Comments-27-4967802.0-0

File: .github/workflows/release.yml (hunk 1 above)
Comment (vzhukovs) [reply .github/workflows/release.yml]: Changed to `docker/login-action@v1`

**Comment by nickboldt** [comment]
> `QUAY_REPOSITORY`

What value should that have, and why does it need to be secret?

**Comment by vzhukovs** [comment]
>What value should that have, and why does it need to be secret?

It's variable that stores the repository under which images will be published (and this secret is used only from GitHub workflow). It is used as a secret, because there is no way to provide environment variable (in GitHub) without storing this information in the file, which is redundant. At least I didn't find such such way in repository settings.

**Pull request merged**